src/main/java/io/confluent/idesidecar/restapi/kafkarest/PartitionManagerImpl.java

```diff
@@ -0,0 +1,39 @@
+package io.confluent.idesidecar.restapi.kafkarest;
+
+import io.smallrye.mutiny.Uni;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import java.util.List;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.common.TopicPartitionInfo;
+import org.apache.kafka.common.errors.UnknownTopicOrPartitionException;
+
+@RequestScoped
+public class PartitionManagerImpl extends Manager {
+  @Inject
+  TopicManagerImpl topicManager;
```

Comment:
```suggestion
  TopicManager topicManager;
```

Do you know if we can inject the interface, `TopicManager`, instead of the class, `TopicManagerImpl`, here? In general, interfaces ensure that components are loosely coupled and might ease mocking in tests.

Reply:
Yes, but I remember encountering some sort of a bean injection issue that's not occurring anymore, maybe I was doing something else wrong. I've made changes across kafkarest classes to inject interfaces wherever possible rather than implementations.